danielMunteanu/src/main/java/danielCollectionTask/FilterByCity.java

```diff
@@ -0,0 +1,20 @@
+package danielCollectionTask;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class FilterByCity {
+    public static List<User> filterUsersByCity(List<User> users, String city) {
+        List<User> userListResult = new ArrayList<>();
+        for (User user : users) {
+            for (Map.Entry<String, Address> entry : user.addresses.entrySet()) {
+                if (entry.getValue().city.equals(city)) {
+                    userListResult.add(user);
+                    System.out.println("Username: " + user.username + ", Type of address = " + entry.getKey() + ", city = " + entry.getValue().city);
```

Comment:
There is no need to print out the user. The method is used only to filter the users list by their city

Reply:
fixed in next commit